packages/client/src/workspaces/contexts/ThothInterfaceProvider.tsx

```diff
@@ -12,18 +12,23 @@ import { usePubSub } from '../../contexts/PubSubProvider'
 import { useFetchFromImageCacheMutation } from '@/state/api/visualGenerationsApi'
 import { useGetSpellQuery, useRunSpellMutation } from '@/state/api/spells'
 import { useAuth } from '@/contexts/AuthProvider'
+import { useDispatch } from 'react-redux'
+import { closeTab } from '@/state/tabs'
+import { useNavigate } from 'react-router'
 
 const Context = createContext<EditorContext>(undefined!)
 
 export const useThothInterface = () => useContext(Context)
 
 const ThothInterfaceProvider = ({ children, tab }) => {
+  const dispatch = useDispatch()
+  const navigate = useNavigate()
   const { events, publish, subscribe } = usePubSub()
   const spellRef = useRef<Spell | null>(null)
   const [fetchFromImageCache] = useFetchFromImageCacheMutation()
   const { user } = useAuth()
   const [_runSpell] = useRunSpellMutation()
-  const { data: _spell } = useGetSpellQuery(
+  const { data: _spell, isError } = useGetSpellQuery(
```

Comment:
We should probably rename this restructure similar to _spell to maintain uniqueness in case we need to check any other errors in the same file. Maybe `isError: getSpellError` or similar?